latch_cli/nextflow/config.py

```diff
@@ -0,0 +1,151 @@
+import json
+from pathlib import Path
+from textwrap import dedent
+from typing import Any, Optional, Type
+
+import click
+
+from latch.types.directory import LatchDir
+from latch.types.file import LatchFile
+from latch_cli.snakemake.config.utils import type_repr
+from latch_cli.snakemake.utils import reindent
+
+
+def get_param_type(details: dict) -> Type:
+    t = details.get("type")
+    if t is None:
+        return Any
+
+    if t == "string":
+        format = details.get("format")
+        if format is not None:
+            if format == "file-path":
+                return LatchFile
+            elif format == "directory-path":
+                return LatchDir
+        return str
+    elif t == "boolean":
+        return bool
+    elif t == "integer":
+        return int
+    elif t == "number":
+        return float
+
+    return Any
+
+
+def generate_metadata(
+    schema_path: Path,
+    *,
+    skip_confirmation: bool = False,
+    generate_defaults: bool = False,
+):
+    schema: dict = json.loads(schema_path.read_text())
+
+    display_name: Optional[str] = schema.get("title")
+    # rahul: seems like it is convention to use "<pipeline-name> pipeline parameters" for schema title
+    suffix = " pipeline parameters"
+    if display_name is not None and display_name.endswith(suffix):
+        display_name = display_name[: -len(suffix)]
+
+    metadata_root = Path("latch_metadata")
+    if metadata_root.is_file():
+        if not click.confirm("A file exists at `latch_metadata`. Delete it?"):
+            raise click.exceptions.Exit(0)
```

Comment:
```suggestion
            raise click.Abort()
```

would abort here bc this is due to user intervention and not bc of an error